### PR TITLE
Fork godep to fix inconsistent abbreviation size

### LINKED ISSUE #70686: Prevent `Comment` field conflicts in Godeps/Godeps.json
**What would you like to be added**:

Prevent `Comment` field conflicts in Godeps/Godeps.json

**Why is this needed**:

`godep` may generate different comments on same commit in same repo. It's annoying when updating dependencies. More discussion: https://github.com/kubernetes/kubernetes/pull/69976#pullrequestreview-167565874

`godep` uses `git describe --tags <commit>` to generate comment, different git versions or local configs will generate different results (latest git use dynamic abbreviation size by default which depends on repo state). And some repo may tag one commit with multiple names which results in multiple comments (depends on your repo state), e.g. 

- https://github.com/go-openapi/analysis/tree/v0.17.1
- https://github.com/go-openapi/analysis/tree/v0.17.2

/kind feature

### PR DESCRIPTION
**Pull request by cblecker**
**What this PR does / why we need it**:
This forks godep instead of vendoring it, and applies a fix to the abbreviation size.

Lots of tiny changes here. Best reviewed on a commit by commit basis.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #70686

**Special notes for your reviewer**:
This is just might be crazy enough to work.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by dims** [comment]
Love it :)

**Comment by neolit123** [comment]
/retest
verify flake.

**Review by neolit123 — APPROVED** [review]
LGTM thanks!

File: hack/lib/util.sh [hunk 1]
```diff
@@ -441,28 +441,35 @@ kube::util::ensure_clean_working_dir() {
...
   fi
 
-  kube::log::status "Installing godep version ${GODEP_VERSION}"
-  go install k8s.io/kubernetes/vendor/github.com/tools/godep/
-  if ! which godep >/dev/null 2>&1; then
-    kube::log::error "Can't find godep - is your GOPATH 'bin' in your PATH?"
-    kube::log::error "  GOPATH: ${GOPATH}"
-    kube::log::error "  PATH:   ${PATH}"
-    return 1
+  # If GODEP is in the path, then use it.
```
[the remark this answers is not in the document]
Comment (cblecker) [reply hack/lib/util.sh]: pretty low. could remove this. 🤷‍♂️

**Review by cofyc — APPROVED** [review]
LGTM

**Comment by cblecker** [comment]
Passing tests. PTAL @thockin :)

**Comment by thockin** [comment]
Should we rename it so there's no ambiguity about which binary is being run?  Maybe not important.  Looking now.

**Comment by cblecker** [comment]
@thockin I changed the "version" to something we can verify and ensure we're running the right version, but we could easily rename the binary too. I'm indifferent.

**Review by thockin — COMMENTED** [review]
/lgtm
/approve

File: hack/godep-save.sh [hunk 2]
```diff
@@ -57,7 +57,6 @@ fi
 REQUIRED_BINS=(
   "github.com/onsi/ginkgo/ginkgo"
   "github.com/jteeuwen/go-bindata/go-bindata"
-  "github.com/tools/godep"
```
[the remark this answers is not in the document]
Comment (cblecker) [reply hack/godep-save.sh]: This list is the "additional go repos to vendor that aren't explicit code dependencies" list. It's not exceptionally well named.

File: hack/lib/util.sh [hunk 3]
```diff
@@ -441,28 +441,30 @@ kube::util::ensure_clean_working_dir() {
...
-
-  if [[ "$(godep version 2>/dev/null)" != *"godep ${GODEP_VERSION}"* ]]; then
-    kube::log::error "Wrong godep version - is your GOPATH 'bin' in your PATH?"
-    kube::log::error "  expected: godep ${GODEP_VERSION}"
-    kube::log::error "  got:      $(godep version)"
-    kube::log::error "  GOPATH: ${GOPATH}"
-    kube::log::error "  PATH:   ${PATH}"
+  # Otherwise, install forked godep
+  kube::log::status "Installing godep version ${godep_target_version}"
+  # Run in hermetic GOPATH
```
[the remark this answers is not in the document]
Comment (cblecker) [reply hack/lib/util.sh]: Not that I've found in my testing. It doesn't impact the CWD or environment once the calling bash script itself exits out (either godep-restore or godep-save).

**Comment by ixdy** [comment]
> Should we rename it so there's no ambiguity about which binary is being run?

kodep? :)

**Comment by neolit123** [comment]
[ANAT (all names are taken)](https://github.com/kodep).

File: hack/lib/util.sh [hunk 4]
```diff
@@ -441,28 +441,30 @@ kube::util::ensure_clean_working_dir() {
...
+  kube::golang::setup_env
```
Comment (ixdy) [inline hack/lib/util.sh]: I think this line is now causing `GOPATH` to  be set to "${KUBE_GOPATH}` (i.e. `${KUBE_ROOT}/_output/local/go`) for the rest of the script invocation.

This means that `hack/godep-restore.sh` and `hack/godep-save.sh` are saving into `_output/local/go` now, which is maybe fine (the general recommendation is to run these under `hack/run-in-gopath.sh`), but it's also unexpected.

cc @apelisse

Comment (cblecker) [reply hack/lib/util.sh]: Yup, you're right. Fix in https://github.com/kubernetes/kubernetes/pull/71538

**Pull request merged**